AssertionError: command count not found in replica metrics

### What version of Materialize are you using?

7593a54d6065

### What is the issue?

Seen in https://buildkite.com/materialize/tests/builds/77894#018e1a45-4ae0-4796-b60c-25b24ce3cd37
```
2024-03-07 18:59:21 UTC	==> mzcompose: test case test-github-15531 failed: builtins.AssertionError: command count not found in replica metrics
2024-03-07 18:59:21 UTC	Traceback (most recent call last):
2024-03-07 18:59:21 UTC	  File "/var/lib/buildkite-agent/builds/buildkite-aarch64-b9f4a11-i-0a246bbb27eea7922-1/materialize/tests/misc/python/materialize/mzcompose/composition.py", line 560, in test_case
2024-03-07 18:59:21 UTC	    yield
2024-03-07 18:59:21 UTC	  File "/var/lib/buildkite-agent/builds/buildkite-aarch64-b9f4a11-i-0a246bbb27eea7922-1/materialize/tests/test/cluster/mzcompose.py", line 82, in workflow_default
2024-03-07 18:59:21 UTC	    c.workflow(name)
2024-03-07 18:59:21 UTC	  File "/var/lib/buildkite-agent/builds/buildkite-aarch64-b9f4a11-i-0a246bbb27eea7922-1/materialize/tests/misc/python/materialize/mzcompose/composition.py", line 453, in workflow
2024-03-07 18:59:21 UTC	    func(self)
2024-03-07 18:59:21 UTC	  File "/var/lib/buildkite-agent/builds/buildkite-aarch64-b9f4a11-i-0a246bbb27eea7922-1/materialize/tests/test/cluster/mzcompose.py", line 341, in workflow_test_github_15531
2024-03-07 18:59:21 UTC	    ) = find_command_history_metrics(c)
2024-03-07 18:59:21 UTC	        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-03-07 18:59:21 UTC	  File "/var/lib/buildkite-agent/builds/buildkite-aarch64-b9f4a11-i-0a246bbb27eea7922-1/materialize/tests/test/cluster/mzcompose.py", line 296, in find_command_history_metrics
2024-03-07 18:59:21 UTC	    assert replica_command_count_found, "command count not found in replica metrics"
2024-03-07 18:59:21 UTC	AssertionError: command count not found in replica metrics
```
ci-regexp: command count not found in replica metrics

--- a/test/cluster/mzcompose.py
+++ b/test/cluster/mzcompose.py
@@ -325,8 +325,15 @@ def workflow_test_github_15531(c: Composition) -> None:
         -- table for slow-path peeks
         CREATE TABLE t2 (a int);
         INSERT INTO t2 VALUES (84);
+
+        -- Wait for the cluster to be ready.
+        SELECT * FROM t;
+        SELECT * FROM t2;
         """
     )
+
+    # Wait a bit to let the metrics refresh.
+    time.sleep(2)
 
     # obtain initial history size and dataflow count
     (
@@ -368,6 +375,9 @@ def workflow_test_github_15531(c: Composition) -> None:
             SELECT * FROM t2;
             """
         )
+
+    # Wait a bit to let the metrics refresh.
+    time.sleep(2)
 
     # Check that history size and dataflow count are well-behaved.
     # Dataflow count can plausibly be more than 1, if compaction is delayed.

